src/main/kotlin/app/cash/licensee/plugin.kt

```diff
@@ -49,76 +51,95 @@ class LicenseePlugin : Plugin<Project> {
+    }
+    project.pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
+      configureJavaPlugin(project)
+    }
+    project.pluginManager.withPlugin("org.jetbrains.kotlin.js") {
+      // The JS plugin uses the same runtime configuration name as the Java plugin.
+      configureJavaPlugin(project)
+    }
 
-      var rootTask: TaskProvider<Task>? = null
-      if (project.plugins.hasPlugin("org.jetbrains.kotlin.js")) {
-        // The JS plugin uses the same runtime configuration name as the Java plugin.
-        configureJavaPlugin(project)
-      } else if (project.plugins.hasPlugin("org.jetbrains.kotlin.multiplatform")) {
-        rootTask = project.tasks.register(baseTaskName) {
-          it.group = VERIFICATION_GROUP
-          it.description = taskDescription("all Kotlin targets")
-        }
-        if (androidPlugin != null) {
-          configureKotlinMultiplatformTargets(project, rootTask, skipAndroid = true)
-          configureAndroidVariants(project, rootTask, androidPlugin, prefix = true)
-        } else {
-          configureKotlinMultiplatformTargets(project, rootTask)
-        }
-      } else if (project.plugins.hasPlugin("java")) {
-        // Note: java-library applies java so we only need to look for the latter.
-        // Note: org.jetbrains.kotlin.jvm applies java so we only need to look for the latter.
-        configureJavaPlugin(project)
-      } else if (androidPlugin != null) {
-        rootTask = project.tasks.register(baseTaskName) {
-          it.group = VERIFICATION_GROUP
-          it.description = taskDescription("all Android variants")
-        }
-        configureAndroidVariants(project, rootTask, androidPlugin)
-      } else {
-        val name = if (project === project.rootProject) {
-          "root project"
-        } else {
-          "project ${project.path}"
-        }
-        throw IllegalStateException(
-          "'app.cash.licensee' requires compatible language/platform plugin to be applied ($name)",
-        )
-      }
+    project.pluginManager.withPlugin("org.jetbrains.kotlin.multiplatform") {
+      configureMultiPlatformPlugin(project)
+    }
 
-      if (rootTask != null) {
-        project.tasks.named(CHECK_TASK_NAME).configure {
-          it.dependsOn(rootTask)
-        }
-      }
+    project.pluginManager.withPlugin("com.android.application") {
+      handleAndroid(project, AndroidPlugin.Application)
+    }
+    project.pluginManager.withPlugin("com.android.library") {
+      handleAndroid(project, AndroidPlugin.Library)
+    }
+    project.pluginManager.withPlugin("com.android.dynamic-feature") {
+      handleAndroid(project, AndroidPlugin.DynamicFeature)
+    }
+  }
+}
+
+private fun handleAndroid(
+  project: Project,
+  appliedAndroidPlugin: AndroidPlugin,
+) {
+  project.afterEvaluate {
+    val withMPP = project.pluginManager.hasPlugin("org.jetbrains.kotlin.multiplatform")
+    if (withMPP) {
+      val rootTask = configureMultiPlatformPlugin(project)
```

Comment:
Calling `configureMultiPlatformPlugin` again is safe, because the call has no side effects.